File: 2020_07_28_Algorithm/src/com/company/FindNum_1920.java
package com.company;
//백준 1920 수 찾기
import java.util.Arrays;
import java.util.Scanner;

public class FindNum_1920 {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        int N = sc.nextInt();
        int[] arrN = new int[N];
        for (int i  = 0; i < N; i++){
            arrN[i] = sc.nextInt();
        }

        int M = sc.nextInt();
        int[] arrM = new int[M];
        for (int i = 0; i < M; i++){
            arrM[i] = sc.nextInt();
        }
        Arrays.sort(arrN);
        for (int i = 0; i < M; i++){
            System.out.println(bin_tree(arrN, arrM[i], 0 , N-1));
        }

    }
    public static int bin_tree(int[] arrN, int i, int start, int end){
        if (start > end) {
            return 0;
        }
        int mid = (start + end)/2;

        if (arrN[mid] > i){
            return bin_tree(arrN, i, start, mid-1);
        }
        else if (arrN[mid] < i){
            return bin_tree(arrN, i, mid+1, end);
        }
        else
            return 1;
    }
}

File: 2020_07_28_Algorithm/python/FindNum_1920.py
# 백준 1920 수 찾기

N = int(input())
arrA = list(map(int, input().split(' ')))

M = int(input())
arrM = list(map(int, input().split(' ')))

arrA.sort()

def bin_tree(arrA, i, start, end): # bin_tree(비교될 배열A, 입력하는 배열M, 0부터, 배열 A의 길이 -1)
    if start > end: #존재하는 수 없음 0 리턴
        return 0
    mid = (start + end) // 2 # 가운데 수 지정
    if arrA[mid] > i: # 배열 A의 가운데 인덱스 와 비교하는 값 비교 후
        return bin_tree(arrA, i, start, mid-1) # 가운데 수보다 적은 수로 돌아가서 반복
    elif arrA[mid] < i:
        return bin_tree(arrA, i, mid+1, end) # 가운데 수보다 큰 수로 돌아가서 반복
    else:
        return 1

for i in arrM:
    print(bin_tree(arrA, i, 0, N-1))





